File: Python/Programmars/Level_1/콜라츠 추측.py
def solution(num):
    answer = 0
    while num > 1:
        if answer == 500:
            return -1
        if num % 2 == 0:
            num = num//2
        else:
            num = num*3 + 1
        answer += 1
    return answer


print(solution(6))  # 8
print(solution(16))  # 4
print(solution(626331))  # -1

File: JavaScript/Programmars/Level_1/콜라츠 추측.js
function solution(num) {
  let answer = 0;
  let current = num;
  while (current !== 1 && answer < 500) {
    if (current % 2 === 0) {
      current = Math.floor(current / 2);
    } else {
      current = current * 3 + 1;
    }
    answer++;
  }

  return answer === 500 ? -1 : answer;
}

console.log(solution(6)); // 8
console.log(solution(16)); // 4
console.log(solution(626331)); // -1
